web_dwc2.py

```diff
@@ -1493,71 +1267,17 @@ def get_klipper_macros(self):
     def get_axes_homed(self):
         #	self.toolhead.get_next_move_time() - self.reactor.monotonic()
         homed = []
-        for rail in self.kinematics.rails:
+        if self.kinematics is not None:
```

Comment:
you did similar thing with toolhead elsewere. afaik kinematics an toolhead never can be none. we are running as klippy object, which get initialized after those basics afaik.

Reply:
when klipper is != ready, kinematics is None and it generate an error

Reply:
okay, i just reported nothingon status_0 earlier. So thats why i did not notice this maybe. thanks.